include/MiniFB_ex_cpp.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#if defined(__cplusplus)
+
+#include <functional>
+#include "MiniFB_ex.h"
+
+template <class T>
+void mfb_active_callback(T *obj, void (T::*method)(eBool));
+
+template <class T>
+void mfb_resize_callback(T *obj, void (T::*method)(int, int));
+
+template <class T>
+void mfb_keyboard_callback(T *obj, void (T::*method)(eKey, eKeyMod, eBool));
+
+template <class T>
+void mfb_char_input_callback(T *obj, void (T::*method)(unsigned int));
+
+template <class T>
+void mfb_mouse_button_callback(T *obj, void (T::*method)(eMouseButton, eKeyMod, eBool));
+
+template <class T>
+void mfb_mouse_move_callback(T *obj, void (T::*method)(int, int));
+
+template <class T>
+void mfb_mouse_scroll_callback(T *obj, void (T::*method)(eKeyMod, float, float));
+
+//-------------------------------------
+// To avoid clumsy hands
+//-------------------------------------
+class Stub {
+    template <class T>
+    friend void mfb_active_callback(T *obj, void (T::*method)(eBool));
+    template <class T>
+    friend void mfb_resize_callback(T *obj, void (T::*method)(int, int));
+    template <class T>
+    friend void mfb_mouse_button_callback(T *obj, void (T::*method)(eMouseButton, eKeyMod, eBool));
+    template <class T>
+    friend void mfb_keyboard_callback(T *obj, void (T::*method)(eKey, eKeyMod, eBool));
+    template <class T>
+    friend void mfb_char_input_callback(T *obj, void (T::*method)(unsigned int));
+    template <class T>
+    friend void mfb_mouse_button_callback(T *obj, void (T::*method)(eMouseButton, eKeyMod, eBool));
+    template <class T>
+    friend void mfb_mouse_move_callback(T *obj, void (T::*method)(int, int));
+    template <class T>
+    friend void mfb_mouse_scroll_callback(T *obj, void (T::*method)(eKeyMod, float, float));
+
+    static void active_stub(eBool isActive);
+    static void resize_stub(int width, int height);
+    static void keyboard_stub(eKey key, eKeyMod mod, eBool isPressed);
+    static void char_input_stub(unsigned int);
+    static void mouse_btn_stub(eMouseButton button, eKeyMod mod, eBool isPressed);
+    static void mouse_move_stub(int x, int y);
+    static void scroll_stub(eKeyMod mod, float deltaX, float deltaY);
+
+    static std::function<void(eBool)>                        m_active;
+    static std::function<void(int, int)>                     m_resize;
+    static std::function<void(eKey, eKeyMod, eBool)>         m_keyboard;
+    static std::function<void(unsigned int)>                 m_char_input;
+    static std::function<void(eMouseButton, eKeyMod, eBool)> m_mouse_btn;
+    static std::function<void(int, int)>                     m_mouse_move;
+    static std::function<void(eKeyMod, float, float)>        m_scroll;

```

Comment:
Is `std::function` really needed here? I think just regular function pointers would do just fine.

Reply:
This is to be able to use C++ classes. std::function can wrap normal or member functions. And I don't wanted to use my own old delegate to do that if it is there in the standard.